File: services/rag/app/utils/model_list.py
"""Utilities for parsing comma-separated model lists from environment variables."""


def parse_model_list(value: str | None) -> list[str]:
    """Parse a comma-separated model list into a list of trimmed, non-empty strings."""
    if not value:
        return []
    return [m.strip() for m in value.split(",") if m.strip()]


def get_first_model(value: str | None) -> str | None:
    """Get the first model from a comma-separated model list."""
    models = parse_model_list(value)
    return models[0] if models else None


def get_first_model_or_raise(value: str | None, var_name: str) -> str:
    """Get the first model from a comma-separated model list, or raise if none available."""
    model = get_first_model(value)
    if not model:
        raise ValueError(f"{var_name} is not set or contains no valid models.")
    return model

File: packages/shared/src/utils/model-list.ts
/**
 * Utilities for parsing comma-separated model lists from environment variables.
 */

/** Parse a comma-separated model list into trimmed, non-empty strings. */
export function parseModelList(value: string | null | undefined): string[] {
  if (!value) {
    return [];
  }
  return value
    .split(',')
    .map((m) => m.trim())
    .filter((m) => m.length > 0);
}

/** Get the first model from a comma-separated model list, or null when empty. */
export function getFirstModel(value: string | null | undefined): string | null {
  const models = parseModelList(value);
  return models.length > 0 ? models[0] : null;
}

/** Get the first model from a comma-separated model list, or throw if none. */
export function getFirstModelOrThrow(
  value: string | null | undefined,
  varName: string,
): string {
  const model = getFirstModel(value);
  if (!model) {
    throw new Error(`${varName} is not set or contains no valid models.`);
  }
  return model;
}
